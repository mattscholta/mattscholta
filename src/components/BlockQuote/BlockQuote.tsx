import * as React from 'react';
import classnames from 'classnames';

import styles from './styles.module.css';

export interface BlockQuoteProps {
  className?: string;
  simple?: boolean;
}

/**
 * @name BlockQuote
 * @description Stateless Component / Pure Function
 */
const BlockQuote: React.FC<BlockQuoteProps> = (props) => {
  const { children, className, simple = false } = props;

  // Styles
  const cssComponent = classnames(styles.component, className);

  return (
    <div className={cssComponent}>
      {!simple && <hr />}
      <blockquote>{children}</blockquote>
      {!simple && <hr />}
    </div>
  );
};

export { BlockQuote };
