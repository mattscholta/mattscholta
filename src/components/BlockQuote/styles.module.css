.component {
  font-style: italic;
  font-weight: var(--font-weight-regular);
  padding: var(--grid-4x) 0;
}

.component blockquote {
  margin: var(--grid-8x) auto var(--grid-6x);
  max-width: 80%;
}

.component blockquote b,
.component blockquote em {
  color: var(--spot-grey-9);
}

.component hr {
  margin: var(--grid-2x) 0;
}

@media (max-width: 768px) {
  .component {
    padding: var(--grid-2x) 0;
  }

  .component blockquote {
    margin: var(--grid-4x) auto var(--grid-2x);
  }
}
