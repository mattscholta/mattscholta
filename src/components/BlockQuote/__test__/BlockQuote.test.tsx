import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';

import { BlockQuote, BlockQuoteProps } from '../BlockQuote';

describe('BlockQuote', () => {
  let props: BlockQuoteProps;
  let wrapper: RenderResult;

  const createWrapper = () => <BlockQuote {...props} />;

  beforeEach(() => {
    props = {};
    wrapper = render(createWrapper());
  });

  it('matches snapshot', () => {
    expect(wrapper.container.childNodes).toMatchSnapshot();
  });
});
