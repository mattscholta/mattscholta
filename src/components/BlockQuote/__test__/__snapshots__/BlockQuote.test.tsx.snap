// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockQuote matches snapshot 1`] = `
NodeList [
  <div
    class="component"
  >
    <hr />
    <blockquote />
    <hr />
  </div>,
]
`;
